dll/kernel32.cpp

```diff
@@ -1537,11 +1734,46 @@ namespace kernel32 {
 		return 0; // fail
 	}
 
+	DWORD WIN_FUNC GetEnvironmentVariableA(LPCSTR lpName, LPSTR lpBuffer, DWORD nSize) {
+		DEBUG_LOG("GetEnvironmentVariableA: %s\n", lpName);
+		const char *value = getenv(lpName);
+		if (!value) {
+			return 0;
+		}
+		unsigned int len = strlen(value);
+		if (nSize == 0) {
+			return len + 1;
+		}
+		if (nSize < len) {
+			return len;
+		}
+		memcpy(lpBuffer, value, len + 1);
+		return len;
+	}
+
 	unsigned int WIN_FUNC SetEnvironmentVariableA(const char *lpName, const char *lpValue) {
 		DEBUG_LOG("SetEnvironmentVariableA: %s=%s\n", lpName, lpValue);
 		return setenv(lpName, lpValue, 1 /* OVERWRITE */);
 	}
 
+	DWORD WIN_FUNC GetEnvironmentVariableW(LPCWSTR lpName, LPWSTR lpBuffer, DWORD nSize) {
+		DEBUG_LOG("GetEnvironmentVariableW: %s\n", wideStringToString(lpName).c_str());
+		const char *value = getenv(wideStringToString(lpName).c_str());
+		if (!value) {
+			return 0;
+		}
+		unsigned int len = strlen(value) * 2;
+		if (nSize == 0) {
+			return len + 1;
+		}
+		if (nSize < len) {
+			return len;
+		}
+		const uint16_t *wideValue = stringToWideString(value);
```

Comment:
memory leak; should we maybe add a version of stringToWideString that returns a vector<uint16_t>?

Reply:
Agreed, or maybe `std::wstring`? Added a `free()` for now

Reply:
wstring is UTF-32 on Linux, alas